File: src/StreamCommunity.Host/ClientApp/src/app/viewer-games/state/enlistment.model.ts
import { EnlistmentState } from '../models/enlistment-state';

export interface Enlistment {
  id: number | string;
  userName: string;
  state: EnlistmentState;
  timestamp: Date;
  sortingNo: number;
}

export function createEnlistment(params: Partial<Enlistment>) {
  return {} as Enlistment;
}

File: StreamCommunity.Api/ViewerGames/EnlistmentModel.cs
using System;
using TwitchCommunity.Domain;

namespace StreamCommunity.Api.ViewerGames
{
    public class EnlistmentModel
    {
        public int Id { get; set; }

        /// <summary>
        /// Gets the username
        /// </summary>
        public string UserName { get; set; }

        /// <summary>
        /// Gets the timestamp the enlistment was created
        /// </summary>
        public DateTime Timestamp { get; set; }

        /// <summary>
        /// Gets the state of the enlistment
        /// </summary>
        public EnlistmentState State { get; set; }
    }
}
